### ISSUE TITLE
What is the incidents report for primus-checks doing here?

### ISSUE THREAD

**Issue by anzosasuke** [BinaryAnalysisPlatform/bap-toolkit#12]
Trying to check for primus-checks. I gave the binary with stack buffer overflow error. Here is the binary
[2.zip](https://github.com/BinaryAnalysisPlatform/bap-toolkit/files/10679438/2.zip)

And again the incidents i got : 
[incidents.zip](https://github.com/BinaryAnalysisPlatform/bap-toolkit/files/10679442/incidents.zip)
 

I am not able to make sense out of it. Did it detect, or if it did, how do i go about understanding the output.( This is my first time using this tool.)
bap-version = 2.6.0-alpha+7d79f7f
using docker from bap-toolkit.

**Comment by anzosasuke** [comment]
Thanks so much @ivg

**Comment by anzosasuke** [comment]
@ivg  so, I am actually trying to detect and evaluate the BAP with BAP-toolkit using the different checks. In the incidents, if i do not get any incidents on the juliet test suite for the Use-after-free as well. 

command i am using is using --recipe=use-after-free. There are no incidents of that what so ever. Does that mean this didn't detect any use-after-error in those benchmark suites??

Also these 
```(incident (value-was-checked 3 5))
(incident (value-was-checked 6 8))
(incident (value-was-checked 6 12))```
I did't get these anywhere..

Furthermore, I got these error when i was running the checks for one of the binary.
```[00:00:00]  disassemble: 100% (5/5)
[00:02:08]  primus-greedy:  99% (820/825)
[00:02:09]  primus-mark-visited:  17% (1292/7304)
[00:02:13]  run:   0% (0/1)
[00:02:08]   multi-task-job:  13% (144/1095)
The pass "run" failed with:
Bitvector.extract: len 0 is negative

```
[xz_s.zip](https://github.com/BinaryAnalysisPlatform/bap-toolkit/files/10710703/xz_s.zip)

**Comment by ivg** [comment]
The bitvector extract issue is definitely a bug, can you please create a separate issue on [bap][1]'s issue tracker?

> command i am using is using --recipe=use-after-free. There are no incidents of that what so ever. Does that mean this didn't detect any use-after-error in those benchmark suites??

When I was developing this check I got 100% detection rate if I exclude C++. So something is very fishy going on here. Can you provide more information, e.g., the resulting binaries, log files, etc?


[1]: https://github.com/BinaryAnalysisPlatform/bap?

**Comment by anzosasuke** [comment]
Oh im sorry about putting primus-checks report here. But it was same error for both the checks use-after-free and primus-checks. I will send it to you asap after i reach home. Thanks @ivg

**Comment by anzosasuke** [comment]
Yes, I have added on the bap about it. [#1579](https://github.com/BinaryAnalysisPlatform/bap/issues/1579#issue-1580342153)

About the Juliet test suite, here are the codes for UAF. NPD works fine but not use-after-free.
[link](https://github.com/anzosasuke/NPD)

And also here is the incidents for the UAF and log files.
incident
[incident.zip](https://github.com/BinaryAnalysisPlatform/bap-toolkit/files/10711285/incident.zip)
Binary
[CWE416_Use_After_Free__malloc_free_long_01.zip](https://github.com/BinaryAnalysisPlatform/bap-toolkit/files/10711287/CWE416_Use_After_Free__malloc_free_long_01.zip)
log file
[log.zip](https://github.com/BinaryAnalysisPlatform/bap-toolkit/files/10711284/log.zip)

Please let me know if you need anything else
